### DISCUSSION TITLE
Regression - Cache-control header (pipe) not sets

### DISCUSSION THREAD

**Discussion by dmalki** [sindresorhus/got#1565] [Q&A] [answered]
#### Describe the bug

 - Node.js version:  v15.2.1
 - OS & version: macOS catalina 10.15.6

Pipe cache-Control header response is not sets in the latest got version, as it used to be in got version 10.7.0

#### Actual behavior
Cache-Control is not sets as it used to be in got version 10.7.0
...

#### Expected behavior
Cache-Control" header = "public, max-age=123456"
...

#### Code to reproduce

const got = require('got');
const chai = require('chai'), expect = chai.expect;

const client = got.extend({
    throwHttpErrors: false,
    timeout:20000,
    retry:2
});
const express = require('express');
const app = express();
function fn(){
    const readStream = client.stream('https://www.youtube.com/watch?v=RimGVSA_38o');
    return readStream;
}

app.get('/', async (req, res) => {
    const readStream = fn(req, res);
    readStream.on("response", ()=> {
        res.setHeader("Cache-Control", "public, max-age=123456");
    }).on('error', (err) => {
        console.log('Error in read stream...', err);
    });

    readStream.pipe(res);
});
app.listen(3000);

got('http://localhost:3000').then(response => {
    expect(response.headers["cache-control"]).to.equal("public, max-age=123456");
    console.log('Succedded');
}).catch(error => {
    console.log(error.response);
});

```js
...
```

{
  "name": "AssertionError",
  "message": "expected 'no-cache, no-store, max-age=0, must-revalidate' to equal 'public, max-age=123456'",
  "showDiff": true,
  "actual": "no-cache, no-store, max-age=0, must-revalidate",
  "expected": "public, max-age=123456",
  "stack": "AssertionError: expected 'no-cache, no-store, max-age=0, must-revalidate' to equal 'public, max-age=123456'\n    at /Users/davidmalki/components/linear-jitp/got.js:29:50\n    at processTicksAndRejections (node:internal/process/task_queues:93:5)"

}
#### Checklist

- [ X] I have read the documentation.
- [ X] I have tried my code with the latest version of Node.js and Got.

**Comment by szmarczak** [comment]
Please format the issue properly.

**Comment by dalebel** [comment]
I can't update the original issue, as I'm not the owner. But I'm pasting a simplified and edited version of the above code here, to reproduce the issue
Works with got got 10.7.0
Fails with got version 11.8.1
I run on node.js 14.15.0 on window 10.

``` js
const got = require('got');
const expect = require('chai').expect;
const express = require('express');
const app = express();

app.get('/', async (req, res) => {
    const readStream = got.stream('https://www.youtube.com/watch?v=RimGVSA_38o');
    readStream.on("response", ()=> {
        res.setHeader("Cache-Control", "public, max-age=123456");
    }).on('error', (err) => {
        console.log('Error in read stream...', err);
    });

    readStream.pipe(res);
});
app.listen(3000);

got('http://localhost:3000').then(response => {
    expect(response.headers["cache-control"]).to.equal("public, max-age=123456");
    console.log('Succedded');
});
```

**Comment by szmarczak** [comment, marked as answer]
This is not a regression. See https://github.com/szmarczak/got/commit/d64298cae7556ec1a3501f72ff51190802867c35

Got passes the headers it has received from the response **after** emitting the `response` event, so you can modify the response object. If you don't want to alter it, you need to listen for the `end` event instead.
